core/templates/services/keyboard-shortcut.service.ts

```diff
@@ -19,66 +19,63 @@ import 'mousetrap';
 
 import { Injectable } from '@angular/core';
 import { downgradeInjectable } from '@angular/upgrade/static';
+import { WindowRef } from 'services/contextual/window-ref.service.ts';
+
 
 @Injectable({
   providedIn: 'root'
 })
 export class KeyboardShortcutService {
+  constructor(
+    private windowRef: WindowRef) {
+  }
   bindExplorationPlayerShortcuts() {
     Mousetrap.bind('s', function() {
       document.getElementById('skipToMainContentId').focus();
-      return false;
     });
 
     Mousetrap.bind('k', function() {
       var previousButton = document.getElementById('backButtonId');
       if (previousButton !== null) {
         previousButton.focus();
       }
-      return false;
     });
 
     Mousetrap.bind('j', function() {
       var nextButton = <HTMLElement>document.querySelector(
-        '.protractor-test-next-button');
-      var continueToNextCardButton = <HTMLElement>document.querySelector(
-        '.protractor-test-continue-to-next-card-button');
+        '.oppia-next-button');
       var continueButton = <HTMLElement>document.querySelector(
-        '.protractor-test-continue-button');
+        '.oppia-learner-confirm-button');
       if (nextButton !== null) {
         nextButton.focus();
       }
-      if (continueToNextCardButton !== null) {
-        continueToNextCardButton.focus();
-      }
       if (continueButton !== null) {
         continueButton.focus();
       }
-      return false;
     });
   }
 
   bindLibraryPageShortcuts() {
     Mousetrap.bind('/', function() {
       var searchBar = <HTMLElement>document.querySelector(
-        '.protractor-test-search-input');
+        '.oppia-search-bar-text-input');
       searchBar.focus();
       return false;
     });
 
     Mousetrap.bind('c', function() {
-      document.getElementById('categoryBar').focus();
-      return false;
+      var categoryBar = <HTMLElement>document.querySelector(
+        '.oppia-category-bar-dropdown-toggle');
+      categoryBar.focus();
     });
 
     Mousetrap.bind('s', function() {
       document.getElementById('skipToMainContentId').focus();
-      return false;
     });
   }
 
   setHref(href) {
-    window.location.href = href;
+    this.windowRef.nativeWindow.location.href = href;
```

Comment:
Use this in other places like line 83/87 etc. to make this file independent of this function.

Reply:
Done, with further information in the comment chain linked above